### ISSUE TITLE
Polymorphic types can't be referred to in function body

### ISSUE THREAD

**Issue by AgentME** [facebook/flow#619]
Here's something that caught me by surprise:

```javascript
/* @flow */

function x<T>(a: T) {
  var b: T = a;
}
```

```
[chris /tmp/a]$ flow check

/private/tmp/a/bar.js:4:10,10: identifier T
Could not resolve name

Found 1 error
```

If the T annotation isn't there on line 4, it checks fine and it seems like Flow infers that it's of type T anyway.

**Comment by samwgoldman** [comment]
I have a PR open that addresses this already over in #571, just waiting on a bit of feedback :)

**Comment by AgentME** [comment]
Oh and this is a direct duplicate of https://github.com/facebook/flow/issues/160. I'll close this.

**Comment by samwgoldman** [comment]
@AgentME just a quick update: the restriction on referencing types is now lifted on the latest master, so your example code will type check now.
